File: Raven.Client.Lightweight/IDisposalNotification.cs
using System;

namespace Raven.Client
{
	/// <summary>
	/// Provide a way for interested party to tell whatever implementers have been disposed
	/// </summary>
	public interface IDisposalNotification : IDisposable
	{
		/// <summary>
		/// Called after dispose is completed
		/// </summary>
		event EventHandler AfterDispose;

		/// <summary>
		/// Whatever the instance has been disposed
		/// </summary>
		bool WasDisposed { get; }
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/IDisposalNotification.java
package net.ravendb.client;

import net.ravendb.abstractions.basic.EventHandler;
import net.ravendb.abstractions.basic.VoidArgs;

/**
 * Provide a way for interested party to tell whatever implementers have been disposed
 */
public interface IDisposalNotification extends AutoCloseable {

  /**
   * Called after dispose is completed
   * @param event
   */
  public void addAfterDisposeEventHandler(EventHandler<VoidArgs> event);

  /**
   * Remove event handler
   * @param event
   */
  public void removeAfterDisposeEventHandler(EventHandler<VoidArgs> event);


  /**
   * Whatever the instance has been disposed
   * @return
   */
  public boolean getWasDisposed();

}
